### ISSUE TITLE
Feature: Audit Logging & API Key Audit Trail

### ISSUE THREAD

**Issue by Armin2708** [Alpaca-Network/gatewayz-backend#1209]
## Feature: Audit Logging & API Key Audit Trail

Comprehensive audit trail for API key usage, admin actions, and security events.

```mermaid
flowchart TD
    subgraph Events["Audited Events"]
        A["API Key Created"] --> L["[AUDIT] API_KEY_CREATED"]
        B["API Key Deleted"] --> L
        C["API Key Used"] --> L
        D["Security Violation"] --> L
        E["Admin Action"] --> L
        F["Rate Limit Exceeded"] --> L
        G["Auth Failure"] --> L
        H["Payment Event"] --> L
    end

    L --> M["Stream Logger (stdout/Sentry)"]
    L --> N["Database: activity_log table"]
    L --> O["Prometheus: security event counters"]
    
    P["GET /audit/api-keys/{id}"] --> Q["Usage history for specific key"]
    R["GET /audit/security-events"] --> S["Security violations list"]
```

### Reliability Concerns
- **Compliance**: Audit logs are legal records — loss = compliance violation
- **Accuracy**: Every event must be logged with correct timestamp and context
- **Security**: Audit logs must be immutable — no UPDATE/DELETE operations
- **Performance**: Audit logging must not add latency to the critical path

### Key Files
- `src/routes/audit.py`, `src/security/security.py` (AuditLogger), `src/db/activity.py`

**Comment by Armin2708** [comment]
## User Stories (4)

- [ ] #1313 — Log all API key lifecycle events with structured audit entries
- [ ] #1314 — Log security violations with warning-level severity
- [ ] #1315 — Ensure audit logging does not add latency to the critical path
- [ ] #1316 — Ensure audit log immutability for compliance
